src/test/java/com/google/sps/SearchTest.java

```diff
@@ -14,17 +14,118 @@
 
 package com.google.sps;
 
+import com.google.sps.data.Tutor;
+import com.google.sps.data.TimeRange;
+import com.google.sps.data.TutorSession;
 import org.junit.Assert;
 import org.junit.Test;
+import org.junit.Before;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
+import static org.mockito.Mockito.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Arrays;
+import com.google.gson.Gson;
+import java.io.*;
+import javax.servlet.http.*;
 
 @RunWith(JUnit4.class)
 public final class SearchTest {
 
+    private SearchServlet servlet;
+
+    @Before
+    public void setUp() {
+        servlet = new SearchServlet();
+    }
+
+    @Test
+    public void doGetReturnsCorrectResponseForHistory() throws IOException {
+        HttpServletRequest request = mock(HttpServletRequest.class);       
+        HttpServletResponse response = mock(HttpServletResponse.class); 
+
+        when(request.getParameter("topic")).thenReturn("history");
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(writer);
+
+        //create the hard coded data
+        servlet.doGet(request, response);
+
+        //verify that getParameter was called
+        verify(request, atLeast(1)).getParameter("topic"); 
+        writer.flush(); // it may not have been flushed yet...
+        List<Tutor> expectedTutorList = Arrays.asList(new Tutor("Kashish Arora", "kashisharora@google.com", new String[]{"Math", "History"}, new TimeRange[]{TimeRange.fromStartToEnd(720, 780), TimeRange.fromStartToEnd(900,1020)}, new TutorSession[]{}));
+        String expected = convertListToJson(expectedTutorList);
```

Comment:
Sam suggested getting rid of the gson helper functions and simply writing String expected = new Gson().toJson(object); and I thought that made things cleaner

Reply:
That works better, thanks!